tests/Database/DatabaseEloquentBelongsToManyTest.php

```diff
@@ -554,6 +554,50 @@ public function testTouchMethodSyncsTimestamps()
         $relation->touch();
     }
 
+    /**
+     * @dataProvider toggleMethodListProvider
+     */
+    public function testToggleMethodTogglesIntermediateTableWithGivenArray($list)
+    {
+        $relation = $this->getMockBuilder('Illuminate\Database\Eloquent\Relations\BelongsToMany')->setMethods(['attach', 'detach'])->setConstructorArgs($this->getRelationArguments())->getMock();
```

Comment:
Wouldn't we be better off with integration tests instead of this aggressive mocking with detailed expectations?